drivers/sensor/bmi270/bmi270.h

```diff
@@ -206,8 +213,46 @@ struct bmi270_data {
 	uint16_t gyr_range;
 };
 
-struct bmi270_dev_config {
+union bmi270_bus {
+#if BMI270_BUS_SPI
+	struct spi_dt_spec spi;
+#endif
+#if BMI270_BUS_I2C
 	struct i2c_dt_spec i2c;
+#endif
+};
+
+typedef int (*bmi270_bus_check_fn)(const union bmi270_bus *bus);
+typedef int (*bmi270_bus_init_fn)(const union bmi270_bus *bus);
+typedef int (*bmi270_reg_read_fn)(const union bmi270_bus *bus,
+				  uint8_t start,
+				  uint8_t *data,
+				  uint16_t len);
+typedef int (*bmi270_reg_write_fn)(const union bmi270_bus *bus,
+				   uint8_t start,
+				   const uint8_t *data,
+				   uint16_t len);
+
+struct bmi270_bus_io {
+	bmi270_bus_check_fn check;
+	bmi270_reg_read_fn read;
+	bmi270_reg_write_fn write;
+	bmi270_bus_init_fn init;
+};
+
+#if BMI270_BUS_SPI
+#define BMI270_SPI_OPERATION (SPI_WORD_SET(8) | SPI_TRANSFER_MSB)
+#define BMI270_SPI_ACC_DELAY_US 2
+extern const struct bmi270_bus_io bmi270_bus_io_spi;
+#endif
+
+#if BMI270_BUS_I2C
+extern const struct bmi270_bus_io bmi270_bus_io_i2c;
+#endif
+
+struct bmi270_dev_config {
+	union bmi270_bus bus;
+	const struct bmi270_bus_io *bus_io;
 };
```

Comment:
why is all this in a header file? is it shared with any other module? please move it to the driver otherwise.